rust/lance-linalg/src/kmeans.rs

```diff
@@ -503,11 +515,13 @@ mod tests {
     use super::*;
 
     use arrow_array::Float32Array;
+    use lance_arrow::*;
 
     #[tokio::test]
     async fn test_train_with_small_dataset() {
         let data = Float32Array::from(vec![1.0, 2.0, 3.0, 4.0]);
-        match KMeans::new(&data, 2, 128, 5).await {
+        let data = FixedSizeListArray::try_new_from_values(data, 2).unwrap();
```

Comment:
Do we still need the old `data`?

Reply:
The old data's ownership is passed inside the `FixedSizeListArray()`, right?

Reply:
Oh I see now. Nevermind.